scripts/generateDocs.js

```diff
@@ -0,0 +1,306 @@
+            const lines = data.split(/\r?\n/);
+            const matches = lines[0].match(/^#+ +(.*)/);
+            if (matches) {
+                html.push(matches[1].replace(/`(.*)`/g, '<code>$1</code>'));
+            } else {
+                html.push(convertToReadableFormat(filename));
+            }
+        }
+        html.push("</li></a>");
+        html.push("\n");
+    }
+
+    try {
+        await (await fs.open(treeViewDir + ".md")).close();
+        await handleFile(path.basename(treeViewDir) + ".md", treeViewDir + ".md");
+    } catch (e) { }
+
+    for (const file of dirents) {
+        if (file.name in mapTable) {
+            if (mapTable[file.name] == "") {
+                continue;
+            }
+        }
+
+        const fullName = treeViewDir + '/' + file.name;
+
+        if (file.isDirectory()) {
+            directories.push(file.name);
+            html.push('<li class="list-expanded"><button class="button-list-can-expand">');
+            html.push(convertToReadableFormat(file.name));
+            html.push("</button></li>\n");
+            html.push('<ul class="standard-padding collapsable">\n');
+            html.push(await processTreeViewLayer(fullName));
+            html.push("</ul>\n");
+        } else if (file.isFile()) {
+            if (!file.name.endsWith(".md")) {
+                continue;
+            } else if (directories.includes(file.name.substr(0, file.name.length - 3))) {
+                continue;
+            }
+            if (is_specs_dir) {
+                await handleFile(file.name, fullName, relativeToRootDomain(fullName, "https://github.com/microsoft/vcpkg/tree/master/"));
+            } else {
+                await handleFile(file.name, fullName);
+            }
+        }
+    }
+
+    return html.join('');
+}
+
+/**
+ * @param {string} treeViewDir
+ */
+async function generateTreeView(treeViewDir) {
+    return "<ul class=docs-navigation>" + await processTreeViewLayer(treeViewDir) + "</ul>\n";
+}
+
+function callshowdown(data, currentPath) {
+    var translate = new showdown.Converter();
+    translate.setFlavor('github');
+    translate.setOption('simpleLineBreaks', false);
+    translate.setOption('emoji', true);
+    translate.setOption('smoothLivePreview', true);
+    translate.setOption('encodeEmails', false);
+
+    var html = translate.makeHtml(data);
+
+    html = html
+        .replace(/&lt;\/code&gt;/g, "</code>")
+        .replace("ports&amp;lt;package&amp;gt;\\", "ports\\&lt;package&gt;\\")
+        .replace(/<br \/>/g, " ")
+        .replace(/Vcpkg/g, "vcpkg")
+        .replace('<a href="mailto:vcpkg@microsoft.com">vcpkg@microsoft.com</a>', '<a href="&#x6d;&#x61;&#105;&#108;&#x74;&#111;&#58;v&#99;&#x70;&#107;&#x67;&#64;&#109;&#x69;&#x63;&#114;&#x6f;&#x73;&#x6f;&#102;&#x74;&#x2e;&#x63;&#x6f;&#x6d;">&#118;&#x63;&#x70;&#x6b;&#103;&#x40;&#109;&#x69;&#x63;&#114;&#111;&#x73;&#111;&#102;&#x74;&#x2e;&#99;&#x6f;&#x6d;</a>')
```

Comment:
It's a bit amusing to try to hide this email alias while having the cleartext email alias in the generation script here 🤷‍♂️